pkg/user/postgres.go

```diff
@@ -19,13 +20,39 @@ func NewPostgresRepo(db *gorm.DB) Repository {
 // GetUserByEmail returns a user by email
 func (r *repo) GetUserByEmail(email string) (*models.User, error) {
 	// TODO: Implement GetUserByEmail
+	var user models.User
+	result := r.DB.Where("email = ?", email).First(&user)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+	if result.RowsAffected == 0 {
+		return nil, nil
+	}
+	return &user, nil
 
-	return nil, nil
+	//return nil, nil
 }
 
 // ReplaceApiKey replaces the api key for a user
 func (r *repo) ReplaceApiKey(UserId *uuid.UUID) (*models.ApiKey, error) {
 	// TODO: Implement ReplaceApiKey
+	apiKeyUUID, err := uuid.NewRandom()
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a new API key object
+	apiKey := &models.ApiKey{
+		UserId:    &apiKeyUUID,
+		BaseModel: models.BaseModel{},
+	}
+
+	// Save the API key to the database
+	if err := r.DB.Save(apiKey).Error; err != nil {
+		return nil, err
+	}
```

Comment:
Unfortunately, The logic is flawed here. We fetch the API via UserId, then update the apiKey by generating it inside BaseModel or set the fetched object ID as new UUID.

UserId is being assigned as ApiKeyID, but it is referenced from UserTable, since UUID is always random, that key will not be present inside the users table. It will never work.

Reply:
I will go through it once and get back to you.